### PR TITLE
Migrate the package to use the new `lakefs` SDK wrapper library

### PR DESCRIPTION
**Pull request by nicholasjng**
Addresses #195.

Progress:

- [x] Migrated `LakeFSFileSystem`
- [x] Migrated `LakeFSFile`
- [x] Deleted `lakefs_spec.config`
- [x] Migrated `conftest`
- [x] Migrated unit tests
- [x] Migrated smoke tests
- [x] Deleted `lakefs_spec.client_helpers`
- [x] Migrated `LakeFSTransaction`

A point of contention is the client initialization. ~I removed the client configuration attributes for now, but they could be added back depending on whether we want to still support them.~
EDIT: The attributes are added back, but the configfile is pending a decision, as stated below.

As for the config file - `lakefs` currently only supports the default location `$HOME/.lakectl.yaml`, so if we take over their behavior, the `configfile` argument is not useful anymore until they allow other files.

### CODE REVIEW

**Comment by nicholasjng** [comment]
Hahaha, this change deletes so many lines that the new coverage falls below the current total coverage target!

File: src/lakefs_spec/spec.py [hunk 1]
```diff
@@ -88,42 +84,43 @@ def __init__(
...
-            host=host or os.getenv("LAKEFS_HOST") or lakectl_config.host,
-            api_key=api_key or os.getenv("LAKEFS_API_KEY"),
-            api_key_prefix=api_key_prefix or os.getenv("LAKEFS_API_KEY_PREFIX"),
-            access_token=access_token or os.getenv("LAKEFS_ACCESS_TOKEN"),
-            username=username or os.getenv("LAKEFS_USERNAME") or lakectl_config.username,
-            password=password or os.getenv("LAKEFS_PASSWORD") or lakectl_config.password,
-            ssl_ca_cert=ssl_ca_cert or os.getenv("LAKEFS_SSL_CA_CERT"),
-        )
+            # TODO: `lakefs` does not consider configfiles other than ~/.lakectl.yaml,
+            #  do we keep any file or do we drop configfile support and defer?
```
Comment (maxmynter) [inline src/lakefs_spec/spec.py:101]: I don't see where other configurations would be necessary. So I'd be in favour of deferment

**Review by maxmynter — APPROVED** [review]
Nice!

File: src/lakefs_spec/spec.py (hunk 1 above)
Comment (nicholasjng) [reply src/lakefs_spec/spec.py]: Removed `lakefs_spec.config` in [71bd77d](https://github.com/aai-institute/lakefs-spec/pull/231/commits/71bd77d2717372b1f4f9ff296aa7ab9dbeff9362).

**Pull request merged**